<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>PICsnATAC: Paired Insertion Counting for snATAC-seq data â€¢ PICsnATAC</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="PICsnATAC: Paired Insertion Counting for snATAC-seq data">
<meta name="description" content="A package for Paired Insertion Counting for snATAC-seq data.">
<meta property="og:description" content="A package for Paired Insertion Counting for snATAC-seq data.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">PICsnATAC</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/vignette-1----PIC-counting-with-10X-Cell-Ranger-output.html">Vignette 1 -- PIC counting with 10X Cell Ranger output</a></li>
    <li><a class="dropdown-item" href="articles/vignette-2----PIC-counting-in-Seurat-workflow.html">Vignette 2 -- PIC counting in Seurat/Signac workflow</a></li>
    <li><a class="dropdown-item" href="articles/vignette-3----PIC-counting-in-ArchR-workflow.html">Vignette 3 -- PIC counting in ArchR workflow</a></li>
    <li><a class="dropdown-item" href="articles/vignette-4----PIC-counting-with-dsc-ATAC-seq-data.html">Vignette 4 -- PIC counting with dsc-ATAC-seq data</a></li>
    <li><a class="dropdown-item" href="articles/vignette-5----DAR-analysis-with-PIC-parametric-framework.html">Vignette 5 -- DAR analysis with PIC parametric framework</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="picsnatac">PICsnATAC<a class="anchor" aria-label="anchor" href="#picsnatac"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>The goal of PICsnATAC is to accurately quantify snATAC-seq data. The two key components are:</p>
<div class="section level4">
<h4 id="id_1-constructing-cell-by-peak-matrix-with-paired-insertion-counting-pic">1. <span style="text-decoration:underline">Constructing cell-by-peak matrix</span> with Paired Insertion Counting (PIC).<a class="anchor" aria-label="anchor" href="#id_1-constructing-cell-by-peak-matrix-with-paired-insertion-counting-pic"></a>
</h4>
<div class="section level5">
<h5 id="motivation">
<strong>Motivation</strong>:<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h5>
<p>There are two common counting approaches for snATAC-seq data, <strong>fragment-based counting</strong> and <strong>insertion-based counting</strong>. They generate inconsistent results (see figure below).</p>
<div style="display: flex;">
<p><img src="reference/figures/README-F1c.svg" width="280" style="margin-right: 10px;"><img src="reference/figures/README-F1d.svg" width="280"></p>
</div>
<p><em>Here, we present PIC as a uniform counting choice for various assays, including 10X Genomics snATAC-seq, sci-ATAC-seq, SNARE-seq, dsc-ATAC-seq, s3-ATAC-seq, and scNANO-ATAC.</em></p>
</div>
</div>
<div class="section level4">
<h4 id="id_2-conducting-differential-accessible-region-dar-identification-analysis-using-the-pic-matrix">2. Conducting <span style="text-decoration:underline">Differential Accessible Region (DAR)</span> identification analysis using the PIC matrix.<a class="anchor" aria-label="anchor" href="#id_2-conducting-differential-accessible-region-dar-identification-analysis-using-the-pic-matrix"></a>
</h4>
<div class="section level5">
<h5 id="motivation-1">
<strong>Motivation</strong>:<a class="anchor" aria-label="anchor" href="#motivation-1"></a>
</h5>
<p>Two insertion events with correct primer configurations are required for one viable fragment. We derived the <strong>probability distribution of fragment count</strong> as a function of insertion rate and derived a <strong>model-based insertion rate estimation framework</strong>. We further applied a likelihood ratio test approach to detect DARs between populations.</p>
<p><img src="reference/figures/README-F1a.png" style="width:85.0%"></p>
<p><em>Building upon our model-based insertion rate estimation framework, this analysis provides a more sensitive test of DAR between cell groups (cell types).</em></p>
</div>
</div>
<div class="section level2">
<h2 id="dependencies">Dependencies<a class="anchor" aria-label="anchor" href="#dependencies"></a>
</h2>
<p>Please install the dependent libraries by running the following codes</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'data.table'</span><span class="op">)</span> <span class="co">## (please make sure it is newer than 1.8)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'IRanges'</span>, <span class="st">'GenomicRanges'</span>,<span class="st">'Rsamtools'</span>, <span class="st">'GenomeInfoDb'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the most up-to-date version of PICsnATAC from <a href="https://github.com/Zhen-Miao/PICsnATAC" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"Zhen-Miao/PICsnATAC"</span><span class="op">)</span></span></code></pre></div>
<p>PIC-snATAC is a light-weight package. Installation typically takes &lt;5 min.</p>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>This is a basic example which shows you how to construct PIC-based cell by peak matrix:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://zhen-miao.github.io/PICsnATAC/">PICsnATAC</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">pic_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/PIC_counting.html">PIC_counting</a></span><span class="op">(</span><span class="va">cells</span>, <span class="va">fragment_tsv_gz_file_location</span>, <span class="va">peak_sets</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vignettes">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes"></a>
</h2>
<p>The vignettes can be found at our <code>pkgdown</code> page:</p>
<p><a href="https://zhen-miao.github.io/PICsnATAC/" class="uri">https://zhen-miao.github.io/PICsnATAC/</a></p>
</div>
<div class="section level2">
<h2 id="interactive-notebooks">Interactive Notebooks<a class="anchor" aria-label="anchor" href="#interactive-notebooks"></a>
</h2>
<p><em>Here, we provided two Google Colab notebooks for users to run the codes on cloud (processed example data and link provided). However, as snATAC-seq data are typically large and computer intensive, it is recommended to run it using computing cluster for large datasets</em></p>
<p>Notebook 1: <a href="https://github.com/Zhen-Miao/PICsnATAC/blob/main/vignettes/Run_PIC_counting_on_pbmc_3k_data.ipynb" class="external-link">Run PIC counting on PBMC data.</a></p>
<p>Notebook 2: <a href="https://github.com/Zhen-Miao/PICsnATAC/blob/main/vignettes/Run_PIC_on_human_brain_data.ipynb" class="external-link">Run PIC DAR on SNARE-seq2 brain data.</a></p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use PICsnATAC in your research, please cite our paper:</p>
<p>Miao Z, and Kim J. <em>Uniform quantification of single-nucleus ATAC-seq data with Paired-Insertion Counting (PIC) and a model-based insertion rate estimator.</em> <em>Nature Methods</em> 21.1 (2024): 32-36.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing PICsnATAC</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Zhen Miao <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3255-9517" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Junhyong Kim <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-7726-8246" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zhen Miao, Junhyong Kim.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
