<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Vignette 2 -- PIC counting in Seurat/Signac workflow • PICsnATAC</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Vignette 2 -- PIC counting in Seurat/Signac workflow">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PICsnATAC</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/vignette-1----PIC-counting-with-10X-Cell-Ranger-output.html">Vignette 1 -- PIC counting with 10X Cell Ranger output</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-2----PIC-counting-in-Seurat-workflow.html">Vignette 2 -- PIC counting in Seurat/Signac workflow</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-3----PIC-counting-in-ArchR-workflow.html">Vignette 3 -- PIC counting in ArchR workflow</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-4----PIC-counting-with-dsc-ATAC-seq-data.html">Vignette 4 -- PIC counting with dsc-ATAC-seq data</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-5----DAR-analysis-with-PIC-parametric-framework.html">Vignette 5 -- DAR analysis with PIC parametric framework</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Vignette 2 -- PIC counting in Seurat/Signac workflow</h1>
            
      

      <div class="d-none name"><code>vignette-2----PIC-counting-in-Seurat-workflow.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="content">content<a class="anchor" aria-label="anchor" href="#content"></a>
</h2>
<p>This Vignette contains an example to use PIC counting in Seurat
workflow.</p>
</div>
<div class="section level2">
<h2 id="required-libraries">required libraries<a class="anchor" aria-label="anchor" href="#required-libraries"></a>
</h2>
<p>Please make sure the following libraries are installed and loaded for
the analysis.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://r-datatable.com" class="external-link">"data.table"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">"GenomicRanges"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://Matrix.R-forge.R-project.org/" class="external-link">"Matrix"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://zhen-miao.github.io/PICsnATAC/">"PICsnATAC"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/stuart-lab/signac" class="external-link">"Signac"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://satijalab.org/seurat" class="external-link">"Seurat"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-files">Input files<a class="anchor" aria-label="anchor" href="#input-files"></a>
</h2>
<p>Input files for PIC counting includes:</p>
<ul>
<li>cell barcodes with metadata (<code>singlecell.csv</code>)</li>
<li>list of peak regions (<code>peaks.bed</code>)</li>
<li>fragment files (<code>fragment.tsv.gz</code>)</li>
<li>fragment file index (<code>fragment.tsv.gz.tbi</code>)</li>
</ul>
<p>Below are the step-by-step instructions for obtaining these input
from 10X Cell Ranger output files. Example datasets can be downloaded
from 10X Genomics website <a href="https://www.10xgenomics.com/resources/datasets/5-k-peripheral-blood-mononuclear-cells-pbm-cs-from-a-healthy-donor-next-gem-v-1-1-1-1-standard-2-0-0" class="external-link">here</a>.</p>
<p>Please also refer to <a href="https://satijalab.org/signac/articles/pbmc_vignette.html" class="external-link">Signac
Vignettes</a> for additional information of Signac workflow</p>
</div>
<div class="section level2">
<h2 id="cell-barcodes">Cell Barcodes<a class="anchor" aria-label="anchor" href="#cell-barcodes"></a>
</h2>
<p>If we want to keep the cells from Cell Ranger filtering scheme:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"atac_pbmc_5k_nextgem_singlecell.csv"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># -- in your own Cell Ranger output, the file name will be 'singlecell.csv'</span></span>
<span><span class="va">meta.data_filtered</span> <span class="op">&lt;-</span> <span class="va">meta.data</span><span class="op">[</span><span class="va">meta.data</span><span class="op">$</span><span class="va">is__cell_barcode</span> <span class="op">==</span> <span class="fl">1</span>, <span class="op">]</span></span>
<span><span class="va">cells</span> <span class="op">&lt;-</span> <span class="va">meta.data_filtered</span><span class="op">$</span><span class="va">barcode</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="peak-set">Peak set<a class="anchor" aria-label="anchor" href="#peak-set"></a>
</h2>
<p>If we want to keep the set of peaks from Cell Ranger:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"atac_pbmc_5k_nextgem_peaks.bed"</span>, header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># -- in your own Cell Ranger output, the file name will be 'peaks.bed'</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"seqname"</span>, <span class="st">"start"</span>, <span class="st">"end"</span><span class="op">)</span></span>
<span><span class="va">peak_sets</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">)</span></span></code></pre></div>
<p>Note, the codes will be the same for MACS2 called peaks, just change
the file name</p>
</div>
<div class="section level2">
<h2 id="run-pic-and-save-output">Run PIC and save output<a class="anchor" aria-label="anchor" href="#run-pic-and-save-output"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fragment_tsv_gz_file_location</span> <span class="op">&lt;-</span> <span class="st">"atac_pbmc_5k_nextgem_fragments.tsv.gz"</span></span>
<span><span class="va">pic_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PIC_counting.html">PIC_counting</a></span><span class="op">(</span></span>
<span>  cells <span class="op">=</span> <span class="va">cells</span>,</span>
<span>  fragment_tsv_gz_file_location <span class="op">=</span> <span class="va">fragment_tsv_gz_file_location</span>,</span>
<span>  peak_sets <span class="op">=</span> <span class="va">peak_sets</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">pic_mat</span>, <span class="st">"PIC_mat_1.rds"</span><span class="op">)</span></span>
<span><span class="co">## if you have multiple matrices, don't forget to save under different names</span></span></code></pre></div>
<p>Note if you have multiple matrices, please make sure you save them
under different file names</p>
</div>
<div class="section level2">
<h2 id="create-seurat-object-with-pic-output">Create Seurat Object with PIC output<a class="anchor" aria-label="anchor" href="#create-seurat-object-with-pic-output"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"PIC_mat_1.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">chrom_assay</span> <span class="op">&lt;-</span> <span class="fu">CreateChromatinAssay</span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="va">counts</span>,</span>
<span>  sep <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">":"</span>, <span class="st">"-"</span><span class="op">)</span>,</span>
<span>  genome <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>  fragments <span class="op">=</span> <span class="va">fragment_tsv_gz_file_location</span>,</span>
<span>  min.cells <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  min.features <span class="op">=</span> <span class="fl">200</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu">CreateSeuratObject</span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="va">chrom_assay</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"peaks"</span>,</span>
<span>  meta.data <span class="op">=</span> <span class="va">metadata</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<p>If you used PIC-snATAC counting in your analysis, please cite our
manuscript:</p>
<p>Miao Z, and Kim J. <em>Uniform quantification of single-nucleus
ATAC-seq data with Paired-Insertion Counting (PIC) and a model-based
insertion rate estimator.</em> <em>Nature Methods</em> 21.1 (2024):
32-36.</p>
<p>Signac manuscript:</p>
<p>Stuart, Tim, et al. “Single-cell chromatin state analysis with
Signac.” <em>Nature methods</em> 18.11 (2021): 1333-1341.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zhen Miao, Junhyong Kim.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
