<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Vignette 3 -- PIC counting in ArchR workflow • PICsnATAC</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Vignette 3 -- PIC counting in ArchR workflow">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PICsnATAC</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/vignette-1----PIC-counting-with-10X-Cell-Ranger-output.html">Vignette 1 -- PIC counting with 10X Cell Ranger output</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-2----PIC-counting-in-Seurat-workflow.html">Vignette 2 -- PIC counting in Seurat/Signac workflow</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-3----PIC-counting-in-ArchR-workflow.html">Vignette 3 -- PIC counting in ArchR workflow</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-4----PIC-counting-with-dsc-ATAC-seq-data.html">Vignette 4 -- PIC counting with dsc-ATAC-seq data</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-5----DAR-analysis-with-PIC-parametric-framework.html">Vignette 5 -- DAR analysis with PIC parametric framework</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Vignette 3 -- PIC counting in ArchR workflow</h1>
            
      

      <div class="d-none name"><code>vignette-3----PIC-counting-in-ArchR-workflow.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="content">content<a class="anchor" aria-label="anchor" href="#content"></a>
</h2>
<p>This Vignette contains an example to use PIC counting in ArchR
workflow.</p>
</div>
<div class="section level2">
<h2 id="required-libraries">required libraries<a class="anchor" aria-label="anchor" href="#required-libraries"></a>
</h2>
<p>Please make sure the following libraries are installed and loaded for
the analysis.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://r-datatable.com" class="external-link">"data.table"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">"GenomicRanges"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://Matrix.R-forge.R-project.org/" class="external-link">"Matrix"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://zhen-miao.github.io/PICsnATAC/">"PICsnATAC"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"ArchR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-files">Input files<a class="anchor" aria-label="anchor" href="#input-files"></a>
</h2>
<p>Below are the step-by-step instructions for obtaining these input.
Here we used the ArchR tutorial dataset to illustrate how to add PIC
counting output to ArchR analysis. Please also refer to <a href="https://www.archrproject.com/articles/Articles/tutorial.html" class="external-link">ArchR
Vignettes</a> for additional information of ArchR workflow</p>
<p>Please make sure to download the tutorial dataset by running
<code>getTutorialData()</code> function:</p>
</div>
<div class="section level2">
<h2 id="load-example-dataset-and-creating-arrow-files">Load Example Dataset and Creating Arrow Files<a class="anchor" aria-label="anchor" href="#load-example-dataset-and-creating-arrow-files"></a>
</h2>
<p>Here we used ArchR example dataset as</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inputFiles</span> <span class="op">&lt;-</span> <span class="fu">getTutorialData</span><span class="op">(</span><span class="st">"Hematopoiesis"</span><span class="op">)</span></span>
<span><span class="co"># inputFiles</span></span>
<span><span class="fu">addArchRGenome</span><span class="op">(</span><span class="st">"hg19"</span><span class="op">)</span></span>
<span><span class="va">ArrowFiles</span> <span class="op">&lt;-</span> <span class="fu">createArrowFiles</span><span class="op">(</span></span>
<span>  inputFiles <span class="op">=</span> <span class="va">inputFiles</span>,</span>
<span>  sampleNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">inputFiles</span><span class="op">)</span>,</span>
<span>  minTSS <span class="op">=</span> <span class="fl">4</span>, <span class="co"># Dont set this too high because you can always increase later</span></span>
<span>  minFrags <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  addTileMat <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  addGeneScoreMat <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="inferring-doublet">Inferring doublet<a class="anchor" aria-label="anchor" href="#inferring-doublet"></a>
</h2>
<p>We have this section as ArchR tutorial recommended. But this is
option and please make sure you understand the process of doublet
inferrence before executing this part</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">doubScores</span> <span class="op">&lt;-</span> <span class="fu">addDoubletScores</span><span class="op">(</span></span>
<span>  input <span class="op">=</span> <span class="va">ArrowFiles</span>,</span>
<span>  k <span class="op">=</span> <span class="fl">10</span>, <span class="co"># Refers to how many cells near a "pseudo-doublet" to count.</span></span>
<span>  knnMethod <span class="op">=</span> <span class="st">"UMAP"</span>, <span class="co"># Refers to the embedding to use for nearest neighbor search.</span></span>
<span>  LSIMethod <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-an-archrproject">Create an ArchRproject<a class="anchor" aria-label="anchor" href="#create-an-archrproject"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">ArchRProject</span><span class="op">(</span></span>
<span>  ArrowFiles <span class="op">=</span> <span class="va">ArrowFiles</span>,</span>
<span>  outputDirectory <span class="op">=</span> <span class="st">"HemeTutorial"</span>,</span>
<span>  copyArrows <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># This is recommened so that you maintain an unaltered</span></span>
<span>  <span class="co">## copy for later usage.</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="clustering-with-iterative-lsi">Clustering with iterative LSI<a class="anchor" aria-label="anchor" href="#clustering-with-iterative-lsi"></a>
</h2>
<p>By default, ArchR will do peak calling with MACS2 for each cluster,
so we will need to run the following functions to generate clusters.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">addIterativeLSI</span><span class="op">(</span>ArchRProj <span class="op">=</span> <span class="va">proj</span>, useMatrix <span class="op">=</span> <span class="st">"TileMatrix"</span>,</span>
<span>                        name <span class="op">=</span> <span class="st">"IterativeLSI"</span><span class="op">)</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">addClusters</span><span class="op">(</span>input <span class="op">=</span> <span class="va">proj</span>, reducedDims <span class="op">=</span> <span class="st">"IterativeLSI"</span><span class="op">)</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">addGroupCoverages</span><span class="op">(</span><span class="va">proj</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="peak-calling-with-macs2">Peak calling with MACS2<a class="anchor" aria-label="anchor" href="#peak-calling-with-macs2"></a>
</h2>
<p>Please make sure MACS2 is installed before running this. ArchR also
supports other peak calling methods, please refer to ArchR manual for
additional options.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathToMacs2</span> <span class="op">&lt;-</span> <span class="fu">findMacs2</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">addReproduciblePeakSet</span><span class="op">(</span></span>
<span>  ArchRProj <span class="op">=</span> <span class="va">proj</span>,</span>
<span>  pathToMacs2 <span class="op">=</span> <span class="va">pathToMacs2</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="obtain-pic-matrix-from-archrproject-input">Obtain PIC matrix from ArchRproject input<a class="anchor" aria-label="anchor" href="#obtain-pic-matrix-from-archrproject-input"></a>
</h2>
<p>Here, we have three samples, we compute the PIC matrix for each of
them separately. Then, they will be loaded to each of the samples.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pic_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>, length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">inputFiles</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">s</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">inputFiles</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co">## select one sample</span></span>
<span>  <span class="va">sample_sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">inputFiles</span><span class="op">)</span><span class="op">[</span><span class="va">s</span><span class="op">]</span></span>
<span>  <span class="va">allCells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu">getCellColData</span><span class="op">(</span><span class="va">proj</span><span class="op">)</span><span class="op">)</span> <span class="co">## all cells after filtering</span></span>
<span>  <span class="va">cells_sample</span> <span class="op">&lt;-</span> <span class="va">allCells</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/startsWith.html" class="external-link">startsWith</a></span><span class="op">(</span><span class="va">allCells</span>, <span class="va">sample_sel</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="va">cells_sample_name_trunc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">".*\\#"</span>, <span class="st">""</span>, <span class="va">cells_sample</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">peak_sets</span> <span class="op">&lt;-</span> <span class="va">proj</span><span class="op">@</span><span class="va">peakSet</span> <span class="co">## the peak set, from MACS2 peak calling results</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">peak_sets</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span> <span class="co">## remove the peakset names</span></span>
<span>  <span class="va">fragment_tsv_gz_file_location</span> <span class="op">&lt;-</span> <span class="va">inputFiles</span><span class="op">[</span><span class="va">s</span><span class="op">]</span></span>
<span></span>
<span>  <span class="va">pic_mat</span><span class="op">[[</span><span class="va">s</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PIC_counting.html">PIC_counting</a></span><span class="op">(</span></span>
<span>    cells <span class="op">=</span> <span class="va">cells_sample_name_trunc</span>,</span>
<span>    fragment_tsv_gz_file_location <span class="op">=</span> <span class="va">fragment_tsv_gz_file_location</span>,</span>
<span>    peak_sets <span class="op">=</span> <span class="va">peak_sets</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="co">## change back the cell names to match with original input</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pic_mat</span><span class="op">[[</span><span class="va">s</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">cells_sample</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">## we can run the following codes to get the complete peak by cell matrix</span></span>
<span><span class="co">## and we may save this matrix for tuture uses (e.g., if we hope to run</span></span>
<span><span class="co">## other analyses)</span></span>
<span></span>
<span><span class="co"># pic_mat_comb &lt;- do.call(cbind, pic_mat)</span></span>
<span><span class="co"># pic_mat_comb &lt;- pic_mat_comb[,rownames(getCellColData(proj))]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="add-pic-matrix-to-archrproject-object">Add PIC matrix to ArchRproject object<a class="anchor" aria-label="anchor" href="#add-pic-matrix-to-archrproject-object"></a>
</h2>
<p>Here we iteratively add PIC matrix to the ArchR object. We have two
options. 1. Overwrite the PeakMatrix. This will change the original
PeakMatrix counts. 2. Create another data slot.</p>
<p>Below, we provide example codes for each setting. ## Overwrite
original PeakMatrix</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## first, save the data.frame</span></span>
<span><span class="va">pks</span> <span class="op">&lt;-</span> <span class="va">proj</span><span class="op">@</span><span class="va">peakSet</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pks</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">pks</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">pks</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">s</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">inputFiles</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="fu">ArchR</span><span class="fu">:::</span><span class="fu">.initializeMat</span><span class="op">(</span></span>
<span>    ArrowFile <span class="op">=</span> <span class="va">ArrowFiles</span><span class="op">[</span><span class="va">s</span><span class="op">]</span>, Group <span class="op">=</span> <span class="st">"PeakMatrix"</span>,</span>
<span>    Class <span class="op">=</span> <span class="st">"integer"</span>, Units <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    cellNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pic_mat</span><span class="op">[[</span><span class="va">s</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    featureDF <span class="op">=</span> <span class="va">pks</span>, params <span class="op">=</span> <span class="st">"PeakMatrix"</span>,</span>
<span>    force <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="fu">ArchR</span><span class="fu">:::</span><span class="fu">.addMatToArrow</span><span class="op">(</span></span>
<span>    mat <span class="op">=</span> <span class="va">pic_mat</span><span class="op">[[</span><span class="va">s</span><span class="op">]</span><span class="op">]</span>,</span>
<span>    ArrowFile <span class="op">=</span> <span class="va">ArrowFiles</span><span class="op">[</span><span class="va">s</span><span class="op">]</span>,</span>
<span>    Group <span class="op">=</span> <span class="st">"PeakMatrix/matrix_content"</span>,</span>
<span>    binarize <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="section level3">
<h3 id="create-another-data-slot-called-pic">Create another data slot called “PIC”<a class="anchor" aria-label="anchor" href="#create-another-data-slot-called-pic"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## first, save the data.frame</span></span>
<span><span class="va">pks</span> <span class="op">&lt;-</span> <span class="va">proj</span><span class="op">@</span><span class="va">peakSet</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pks</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">pks</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">pks</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">s</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">inputFiles</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="fu">ArchR</span><span class="fu">:::</span><span class="fu">.initializeMat</span><span class="op">(</span></span>
<span>    ArrowFile <span class="op">=</span> <span class="va">ArrowFiles</span><span class="op">[</span><span class="va">s</span><span class="op">]</span>, Group <span class="op">=</span> <span class="st">"PIC"</span>,</span>
<span>    Class <span class="op">=</span> <span class="st">"integer"</span>, Units <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    cellNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pic_mat</span><span class="op">[[</span><span class="va">s</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    featureDF <span class="op">=</span> <span class="va">pks</span>, params <span class="op">=</span> <span class="st">"PIC"</span>,</span>
<span>    force <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="fu">ArchR</span><span class="fu">:::</span><span class="fu">.addMatToArrow</span><span class="op">(</span></span>
<span>    mat <span class="op">=</span> <span class="va">pic_mat</span><span class="op">[[</span><span class="va">s</span><span class="op">]</span><span class="op">]</span>,</span>
<span>    ArrowFile <span class="op">=</span> <span class="va">ArrowFiles</span><span class="op">[</span><span class="va">s</span><span class="op">]</span>,</span>
<span>    Group <span class="op">=</span> <span class="st">"PIC/matrix_content"</span>,</span>
<span>    binarize <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="get-available-matrices">Get available Matrices<a class="anchor" aria-label="anchor" href="#get-available-matrices"></a>
</h2>
<p>The function below will display available cell-by-feature matrices
you have.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getAvailableMatrices</span><span class="op">(</span><span class="va">proj</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="save-and-load-archrproject">Save and load ArchRProject<a class="anchor" aria-label="anchor" href="#save-and-load-archrproject"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## to save object</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">saveArchRProject</span><span class="op">(</span>ArchRProj <span class="op">=</span> <span class="va">proj</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## to load object</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">loadArchRProject</span><span class="op">(</span>path <span class="op">=</span> <span class="st">"HemeTutorial"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<p>If you used PIC-snATAC counting in your analysis, please cite our
manuscript:</p>
<p>Miao Z, and Kim J. <em>Uniform quantification of single-nucleus
ATAC-seq data with Paired-Insertion Counting (PIC) and a model-based
insertion rate estimator.</em> <em>Nature Methods</em> 21.1 (2024):
32-36.</p>
<p>ArchR manuscript:</p>
<p>Granja, Jeffrey M., et al. “ArchR is a scalable software package for
integrative single-cell chromatin accessibility analysis.” <em>Nature
genetics</em> 53.3 (2021): 403-411.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zhen Miao, Junhyong Kim.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
