<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Vignette 5 -- DAR analysis with PIC parametric framework â€¢ PICsnATAC</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Vignette 5 -- DAR analysis with PIC parametric framework">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PICsnATAC</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/vignette-1----PIC-counting-with-10X-Cell-Ranger-output.html">Vignette 1 -- PIC counting with 10X Cell Ranger output</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-2----PIC-counting-in-Seurat-workflow.html">Vignette 2 -- PIC counting in Seurat/Signac workflow</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-3----PIC-counting-in-ArchR-workflow.html">Vignette 3 -- PIC counting in ArchR workflow</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-4----PIC-counting-with-dsc-ATAC-seq-data.html">Vignette 4 -- PIC counting with dsc-ATAC-seq data</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-5----DAR-analysis-with-PIC-parametric-framework.html">Vignette 5 -- DAR analysis with PIC parametric framework</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Vignette 5 -- DAR analysis with PIC parametric framework</h1>
            
      

      <div class="d-none name"><code>vignette-5----DAR-analysis-with-PIC-parametric-framework.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="content">Content<a class="anchor" aria-label="anchor" href="#content"></a>
</h2>
<p>This Vignette contains an example to conduct DAR analysis with PIC
parametric framework. We developed a statistical framework to test the
differential accessible regions (DARs) between cell types. Our framework
incorporated the distribution of snATAC-seq count data (PIC count, can
also run on fragment count), and will have a higher power in detecting
DARs in snATAC-seq data</p>
</div>
<div class="section level2">
<h2 id="required-libraries">Required libraries<a class="anchor" aria-label="anchor" href="#required-libraries"></a>
</h2>
<p>Please make sure the following libraries are installed and loaded for
the analysis.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://r-datatable.com" class="external-link">"data.table"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">"GenomicRanges"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://Matrix.R-forge.R-project.org/" class="external-link">"Matrix"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://zhen-miao.github.io/PICsnATAC/">"PICsnATAC"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-files">Input files<a class="anchor" aria-label="anchor" href="#input-files"></a>
</h2>
<p>The PIC test framework requires the following input:</p>
<ul>
<li>peak by cell PIC count matrix (<code>pic_mat</code>)</li>
<li>cell type labels (<code>cell_types</code>)</li>
</ul>
<p>Note, please refer to our vignettes for obtaining
<code>pic_mat</code>, and clustering and annotation can be conducted
with Seurat, ArchR, or other pipelines. Below, we describe the two steps
of the test</p>
</div>
<div class="section level2">
<h2 id="cell-specific-capturing-rate-estimation">Cell-specific capturing rate estimation<a class="anchor" aria-label="anchor" href="#cell-specific-capturing-rate-estimation"></a>
</h2>
<p>As described in the manuscript, in the snATAC-seq data, each cell may
have largely variant capturing rate (sequencing depth), and here, we
will compute the caputuring rate in each cell while separating the
effect of cell type-specific open probability</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">r_by_ct_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_r_by_ct_mat_pq.html">get_r_by_ct_mat_pq</a></span><span class="op">(</span></span>
<span>  cell_type_set <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">cell_types</span><span class="op">)</span>,</span>
<span>  r_by_c <span class="op">=</span> <span class="va">pic_mat</span>,</span>
<span>  cell_type_labels <span class="op">=</span> <span class="va">cell_types</span>,</span>
<span>  n_features_per_cell <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">pic_mat</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The object <code>r_by_ct_out</code> is a list with two elements,</p>
<ul>
<li>
<strong>p_by_t</strong> Peak by cell type matrix, each element
represents the open probability of the peak in the corresponding cell
type</li>
<li>
<strong>q_vec</strong> A vector of cell-specific capturing rate</li>
</ul>
<p>The second element will be the input of the next function</p>
</div>
<div class="section level2">
<h2 id="dar-test">DAR test<a class="anchor" aria-label="anchor" href="#dar-test"></a>
</h2>
<p>With the cell-specific caputuring rate, we can now execute our DAR
test function</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p_vals</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DAR_by_LRT.html">DAR_by_LRT</a></span><span class="op">(</span>pic_mat <span class="op">=</span> <span class="va">pic_mat</span>, capturing_rates <span class="op">=</span> <span class="va">r_by_ct_out</span><span class="op">$</span><span class="va">q_vec</span>, cell_type_labels <span class="op">=</span> <span class="va">cell_types</span><span class="op">)</span></span></code></pre></div>
<p>The output is a p value vector, specifying the p value for each peak.
In addition, you can calculate the mean log fold change after adjusting
for the capturing rate to add additional criteria for determining the
set of DAR. The example code is provided below:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_type_set</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">cell_types</span><span class="op">)</span></span>
<span><span class="va">capturing_rates</span> <span class="op">&lt;-</span> <span class="va">r_by_ct_out</span><span class="op">$</span><span class="va">q_vec</span></span>
<span></span>
<span><span class="co">## mean accessibility values after correcting for sequencing depth</span></span>
<span><span class="va">rm1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">pic_mat</span><span class="op">[</span>, <span class="va">cell_types</span> <span class="op">==</span> <span class="va">cell_type_set</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://rdrr.io/r/base/matmult.html" class="external-link">%*%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">/</span> <span class="va">capturing_rates</span><span class="op">[</span><span class="va">cell_types</span> <span class="op">==</span> <span class="va">cell_type_set</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">cell_types</span> <span class="op">==</span> <span class="va">cell_type_set</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">rm2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">pic_mat</span><span class="op">[</span>, <span class="va">cell_types</span> <span class="op">==</span> <span class="va">cell_type_set</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://rdrr.io/r/base/matmult.html" class="external-link">%*%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">/</span> <span class="va">capturing_rates</span><span class="op">[</span><span class="va">cell_types</span> <span class="op">==</span> <span class="va">cell_type_set</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">cell_types</span> <span class="op">==</span> <span class="va">cell_type_set</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## sequencing depth corrected log fold change</span></span>
<span><span class="va">log_fc_depth_corrected</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">rm2</span> <span class="op">+</span> <span class="fl">0.0001</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">rm1</span> <span class="op">+</span> <span class="fl">0.0001</span><span class="op">)</span> <span class="co"># 0.0001 is</span></span>
<span><span class="co">## added to prevent log(0)</span></span>
<span></span>
<span><span class="co">## in situations sequencing depth disparity is not a concern, we can also</span></span>
<span><span class="co">## use the uncorrected mean accessibility value, the codes are</span></span>
<span></span>
<span><span class="va">rm1_unc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">pic_mat</span><span class="op">[</span>, <span class="va">cell_types</span> <span class="op">==</span> <span class="va">cell_type_set</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">rm2_unc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">pic_mat</span><span class="op">[</span>, <span class="va">cell_types</span> <span class="op">==</span> <span class="va">cell_type_set</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">log_fc_uncorrected</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">rm2_unc</span> <span class="op">+</span> <span class="fl">0.0001</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">rm1_unc</span> <span class="op">+</span> <span class="fl">0.0001</span><span class="op">)</span> <span class="co"># 0.0001</span></span>
<span><span class="co">## is added to prevent log(0)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<p>If you used PIC-snATAC counting in your analysis, please cite our
manuscript:</p>
<p>Miao Z, and Kim J. <em>Uniform quantification of single-nucleus
ATAC-seq data with Paired-Insertion Counting (PIC) and a model-based
insertion rate estimator.</em> <em>Nature Methods</em> 21.1 (2024):
32-36.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zhen Miao, Junhyong Kim.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
