<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Vignette 4 -- PIC counting with dsc-ATAC-seq data â€¢ PICsnATAC</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Vignette 4 -- PIC counting with dsc-ATAC-seq data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PICsnATAC</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/vignette-1----PIC-counting-with-10X-Cell-Ranger-output.html">Vignette 1 -- PIC counting with 10X Cell Ranger output</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-2----PIC-counting-in-Seurat-workflow.html">Vignette 2 -- PIC counting in Seurat/Signac workflow</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-3----PIC-counting-in-ArchR-workflow.html">Vignette 3 -- PIC counting in ArchR workflow</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-4----PIC-counting-with-dsc-ATAC-seq-data.html">Vignette 4 -- PIC counting with dsc-ATAC-seq data</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-5----DAR-analysis-with-PIC-parametric-framework.html">Vignette 5 -- DAR analysis with PIC parametric framework</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Vignette 4 -- PIC counting with dsc-ATAC-seq data</h1>
            
      

      <div class="d-none name"><code>vignette-4----PIC-counting-with-dsc-ATAC-seq-data.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="content">content<a class="anchor" aria-label="anchor" href="#content"></a>
</h2>
<p>This Vignette contains an example to use PIC counting with
dsc-ATAC-seq data. Specifically, this data type has more fragments per
cell, making it more quantitative. However, due to bead overloading, we
recommend setting deduplicate step during PIC counting step. Please see
details below.</p>
</div>
<div class="section level2">
<h2 id="required-libraries">required libraries<a class="anchor" aria-label="anchor" href="#required-libraries"></a>
</h2>
<p>Please make sure the following libraries are installed and loaded for
the analysis.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://r-datatable.com" class="external-link">"data.table"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">"GenomicRanges"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://Matrix.R-forge.R-project.org/" class="external-link">"Matrix"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://zhen-miao.github.io/PICsnATAC/">"PICsnATAC"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-files">Input files<a class="anchor" aria-label="anchor" href="#input-files"></a>
</h2>
<p>Below are the step-by-step instructions for obtaining these input.
Example datasets can be downloaded from GEO: GSE123576 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE123576" class="external-link">here</a>.
Here in particular, we used the Mouse brain sample (Mouse #2, Channel
#1).</p>
<p>Please make sure to download the following files for PIC
counting:</p>
<ul>
<li>cell barcodes with metadata
(<code>GSE123576_mousebrain_cellData_revision.tsv</code>)</li>
<li>list of peak regions
(<code>GSE123576_mousebrain_peaks_revision.bed</code>)</li>
<li>fragment files
(<code>GSM3507349_Mouse2-Channel1.fragments.tsv.gz</code>)</li>
<li>fragment file index
(<code>GSM3507349_Mouse2-Channel1.fragments.tsv.gz.tbi</code>)</li>
</ul>
</div>
<div class="section level2">
<h2 id="cell-barcodes">Cell Barcodes<a class="anchor" aria-label="anchor" href="#cell-barcodes"></a>
</h2>
<p>If we want to keep the cells from the publications:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"GSE123576_mousebrain_cellData_revision.tsv"</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"_B.*"</span>, <span class="st">""</span>, <span class="va">cm</span><span class="op">$</span><span class="va">DropBarcode</span><span class="op">)</span></span>
<span><span class="va">cm</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="va">samples</span></span>
<span></span>
<span><span class="co">## choose sample</span></span>
<span><span class="va">cmtsample</span> <span class="op">&lt;-</span> <span class="va">cm</span><span class="op">[</span><span class="va">cm</span><span class="op">$</span><span class="va">sample</span> <span class="op">==</span> <span class="st">"N701_Exp119_sample1_S1"</span>, <span class="op">]</span></span>
<span><span class="va">cells</span> <span class="op">&lt;-</span> <span class="va">cmtsample</span><span class="op">$</span><span class="va">DropBarcode</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="peak-set">Peak set<a class="anchor" aria-label="anchor" href="#peak-set"></a>
</h2>
<p>If we want to keep the set of peaks from the publication:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">feature_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"GSE123576_mousebrain_peaks_revision.bed"</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">feature_df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"seqname"</span>, <span class="st">"start"</span>, <span class="st">"end"</span><span class="op">)</span></span>
<span><span class="va">peak_sets</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">feature_df</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-pic-and-save-output">Run PIC and save output<a class="anchor" aria-label="anchor" href="#run-pic-and-save-output"></a>
</h2>
<p>Please note, for dsc-ATAC-seq data, because the beads are overloaded,
we recommend to set <strong>deduplicate = TRUE</strong> so that we do
not double count fragments. Although, based on our analysis, the
fragment.tsv.gz file provided in GEO has already been de-duplicated, we
still recommend doing so for all dsc-ATAC-seq data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fragment_tsv_gz_file_location</span> <span class="op">&lt;-</span> <span class="st">"GSM3507349_Mouse2-Channel1.fragments.tsv.gz"</span></span>
<span><span class="va">pic_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PIC_counting.html">PIC_counting</a></span><span class="op">(</span></span>
<span>  cells <span class="op">=</span> <span class="va">cells</span>,</span>
<span>  fragment_tsv_gz_file_location <span class="op">=</span> <span class="va">fragment_tsv_gz_file_location</span>,</span>
<span>  peak_sets <span class="op">=</span> <span class="va">peak_sets</span>,</span>
<span>  deduplicate <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">pic_mat</span>, <span class="st">"PIC_mat_1.rds"</span><span class="op">)</span></span>
<span><span class="co">## if you have multiple matrices, don't forget to save under different names</span></span></code></pre></div>
<p>Note if you have multiple matrices, please make sure you save them
under different file names</p>
</div>
<div class="section level2">
<h2 id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<p>If you used PIC-snATAC counting in your analysis, please cite our
manuscript:</p>
<p>Miao Z, and Kim J. <em>Uniform quantification of single-nucleus
ATAC-seq data with Paired-Insertion Counting (PIC) and a model-based
insertion rate estimator.</em> <em>Nature Methods</em> 21.1 (2024):
32-36.</p>
<p>The dsc-ATAC-seq dataset is associated with the following
publication:</p>
<p>Lareau CA, Duarte FM, Chew JG, Kartha VK et al.Â Droplet-based
combinatorial indexing for massive-scale single-cell chromatin
accessibility. <em>Nat Biotechnol</em>. 2019 Aug;37(8):916-924.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zhen Miao, Junhyong Kim.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
